srcdeps-maven-enforcer/src/main/java/org/srcdeps/mvn/enforcer/SrcdepsEnforcer.java

```diff
@@ -56,8 +57,105 @@
 
     private static final Logger log = LoggerFactory.getLogger(SrcdepsEnforcer.class);
 
-    private static void assertNotSrcdeps(List<Dependency> deps, String[] violation)
-            throws LifecycleExecutionException {
+    /**
+     * Checks for failure triggering conditions in the given {@code goals}, {@code profiles} and {@code properties}.
+     *
+     * @param failWith
+     *            the srcdeps configuration node
+     * @param goals
+     *            the goals of the current Maven invocation
+     * @param profiles
+     *            the profiles active in the current Maven invocation
+     * @param properties
+     *            the properties present the current Maven invocation
+     * @return the failure triggering item
+     */
+    static String[] assertFailWith(MavenAssertions failWith, List<String> goals, List<String> profiles,
+            Properties properties) {
+
+        log.debug("Srcdeps Enforcer using failWith {}", failWith);
+        final Set<String> failWithGoals = failWith.getGoals();
+        for (String goal : goals) {
```

Comment:
What's the source level for this repo? You could replace this code with a beautiful lambda if you are able to use Java 8 ;)